contrib/compile-tests.sh

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env bash
+#
+# Checks intended compile failures.
+
+set -euo pipefail
+
+REPO_DIR=$(git rev-parse --show-toplevel)
+
+should_pass() {
+    all_passed=true
+    # All crates in the subdirectory pass/ must compile
+    for dir in pass/*; do
+        if [ -d "$dir" ]; then
+            pushd "$dir" > /dev/null
+            if ! cargo run > /dev/null 2>&1; then
+                echo "compile-tests/$dir/ failed to compile"
+                all_passed=false
+            fi
+            popd > /dev/null
+        fi
+    done
+    if [ "$all_passed" = true ]; then
+        return 0
+    else
+        return 1
```

Comment:
Perhaps just define `result = 0` at the beginning and assign it to `1` on fail and just `return "$result"` to get rid of if.

Reply:
Thanks, this is a nicer way of doing it. I have changed it as suggested.